packages/zipkin/src/batch-recorder.js

```diff
@@ -99,6 +99,18 @@ class BatchRecorder {
           span.delegate.setShared(id.parentId !== id.spanId);
           span.delegate.setKind('CLIENT');
           break;
+        case 'MessageSend':
```

Comment:
Sorry you are right that this doesn't match properly to annotation approach.

I recommend adding the top-level annotations/events ProducerStart/Stop ConsumerStart/Stop similar to here:

https://github.com/openzipkin/zipkin4net/pull/151

Then, when actually writing the data to zipkin v1 format, you'd translate like you have now.

Reply:
so specifically take out 'MessageSend' 'WireSend' 'WireRecv' 'MessageRecv' from here and replace with 'ProducerStart' 'ProducerStop' etc (make same change in annotation.js)

after that please make unit test like exist for other stuff then we should be good!